source/indices.d.ts

```diff
@@ -1,24 +1,21 @@
 /**
- * Generates a tuple based on the length of the input array
- * and fills each index with the index number.
- *
- * @example type FruitIndices = Indices<['apple', 'banana', 'plum']> // [0, 1, 2]
- * @example type FruitIndex = Indices<['apple', 'banana', 'plum']>[number] // 0 | 1 | 2
- *
- * @category Array
- * @category Utility
- */
-export type Indices<
-	Tuple extends any[] | readonly any[],
-	Accumulator extends any[] = [],
-> = LengthOf<Accumulator> extends LengthOf<Tuple>
-	? Accumulator
-	: Indices<Tuple, [...Accumulator, LengthOf<Accumulator>]>;
+Generates a tuple based on the length of the input array and fills each index with the index number.
 
-/**
- * Returns the length property of the input array; tuples will return the number of items in the tuple.
- *
- * @example type TotalFruits = LengthOf<['apple', 'banana', 'plum']> // 3
- * @example type TotalStrings = LengthOf<string[]> // number
- */
-type LengthOf<Tuple extends any[] | readonly any[]> = Tuple['length'];
+@example
+```
+type FruitIndices = Indices<['apple', 'banana', 'plum']> // [0, 1, 2]
+```
+
+Can be used to create an index union (`0 | 1 | 2 | ...`)
+
+@example
+```
+type FruitIndex = Indices<['apple', 'banana', 'plum']>[number] // 0 | 1 | 2
+```
+
+@category Array
+@category Utility
+*/
+export type Indices<T extends unknown[] | readonly unknown[]> = {
```

Comment:
maybe this would be more proper?

```ts
type Indices<T extends unknown[] | readonly unknown[]> = Exclude<
  keyof T,
  keyof unknown[]
> extends `${infer Index extends number}`
  ? Index
  : never
  ```

Reply:
I thought coercing the index to a number would be useful, but in the places I have used it, I think it is easier to work with when left as a string. I'll make another example highlighting what I'm talking about

Reply:
here's that example
```ts
type Zip<Keys extends Record<Indices<Values>, PropertyKey>, Values extends readonly unknown[]> = {
  [K in Indices<Values> as Keys[K]]: Values[K]
}

type Keys = ['a', 'b']
type Values = ['apple', 'butter']

type Zipped = Zip<Keys, Values>
// { a: "apple"; b: "butter"; }
```

Reply:
> maybe this would be more proper?

Why would it be more proper?

Reply:
> I thought coercing the index to a number would be useful, but in the places I have used it, I think it is easier to work with when left as a string. I'll make another example highlighting what I'm talking about

It should be a number. We could have a utility type to turn the numbers to strings.